scilpy/reconst/multi_processes.py

```diff
@@ -472,3 +472,83 @@ def convert_sh_basis(shm_coeff, sphere, mask=None,
     shm_coeff_array[mask] = tmp_shm_coeff_array
 
     return shm_coeff_array
+
+
+def convert_sh_to_sf_parallel(args):
+    sh = args[0]
+    B_in = args[1]
+    new_output_dim = args[2]
+    chunk_id = args[3]
+    sf = np.zeros((sh.shape[0], new_output_dim), dtype=np.float32)
+
+    for idx in range(sh.shape[0]):
+        if sh[idx].any():
+            sf[idx] = np.dot(sh[idx], B_in)
+
+    return chunk_id, sf
+
+
+def convert_sh_to_sf(shm_coeff, sphere, mask=None,
```

Comment:
Since DIPY 's `sh_to_sf_matrix` supports it, maybe we could have an argument for `full_basis`?